flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/EGBasedInputsLocationsRetrieverTest.java

```diff
@@ -0,0 +1,134 @@
+import org.apache.flink.runtime.executiongraph.TestingSlotProvider;
+import org.apache.flink.runtime.executiongraph.restart.NoRestartStrategy;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionType;
+import org.apache.flink.runtime.jobgraph.DistributionPattern;
+import org.apache.flink.runtime.jobgraph.JobVertex;
+import org.apache.flink.runtime.jobmaster.TestingLogicalSlot;
+import org.apache.flink.runtime.jobmaster.slotpool.SlotProvider;
+import org.apache.flink.runtime.scheduler.strategy.ExecutionVertexID;
+import org.apache.flink.runtime.taskmanager.TaskManagerLocation;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.empty;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for {@link EGBasedInputsLocationsRetriever}.
+ */
+public class EGBasedInputsLocationsRetrieverTest extends TestLogger {
+
+	/**
+	 * Tests that can get the producers of consumed result partitions.
+	 */
+	@Test
+	public void testGetConsumedResultPartitionsProducers() throws Exception {
+		final JobVertex producer1 = ExecutionGraphTestUtils.createNoOpVertex(1);
+		final JobVertex producer2 = ExecutionGraphTestUtils.createNoOpVertex(1);
+		final JobVertex consumer = ExecutionGraphTestUtils.createNoOpVertex(1);
+		consumer.connectNewDataSetAsInput(producer1, DistributionPattern.ALL_TO_ALL, ResultPartitionType.PIPELINED);
+		consumer.connectNewDataSetAsInput(producer2, DistributionPattern.ALL_TO_ALL, ResultPartitionType.PIPELINED);
+
+		final ExecutionGraph eg = ExecutionGraphTestUtils.createSimpleTestGraph(new JobID(), producer1, producer2, consumer);
+		final EGBasedInputsLocationsRetriever inputsLocationsRetriever = new EGBasedInputsLocationsRetriever(eg);
+
+		ExecutionVertexID evIdOfProducer1 = new ExecutionVertexID(producer1.getID(), 0);
+		ExecutionVertexID evIdOfProducer2 = new ExecutionVertexID(producer2.getID(), 0);
+		ExecutionVertexID evIdOfConsumer = new ExecutionVertexID(consumer.getID(), 0);
+
+		Collection<Collection<ExecutionVertexID>> producersOfProducer1 =
+				inputsLocationsRetriever.getConsumedResultPartitionsProducers(evIdOfProducer1);
+		Collection<Collection<ExecutionVertexID>> producersOfProducer2 =
+				inputsLocationsRetriever.getConsumedResultPartitionsProducers(evIdOfProducer2);
+		Collection<Collection<ExecutionVertexID>> producersOfConsumer =
+				inputsLocationsRetriever.getConsumedResultPartitionsProducers(evIdOfConsumer);
+
+		assertThat(producersOfProducer1, is(empty()));
+		assertThat(producersOfProducer2, is(empty()));
+		assertThat(producersOfConsumer, contains(Arrays.asList(evIdOfProducer1), Arrays.asList(evIdOfProducer2)));
```

Comment:
`Arrays.asList` with only one argument could be replaced with `Collections.singletonList`

Reply:
This line causes a warning
![image](https://user-images.githubusercontent.com/1681921/59565275-728b6700-9051-11e9-803c-831e27f41f9b.png)
Also see: http://www.tothenew.com/blog/why-is-generic-array-creation-not-allowed-in-java/

Hence, I would propose:
```
		assertThat(producersOfConsumer, hasSize(2));
		assertThat(producersOfConsumer, hasItem(Collections.singletonList(evIdOfProducer1)));
		assertThat(producersOfConsumer, hasItem(Collections.singletonList(evIdOfProducer2)));
```

Reply:
OK